### PR TITLE
HSEARCH-4001 Trigger reindexing when a property is modified, but not when the targeted entity is modified

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4001

### CODE REVIEW

File: documentation/src/main/asciidoc/reference/mapper-orm-mapping-reindexing.asciidoc [hunk 1]
```diff
@@ -116,37 +116,148 @@ that whenever the list of authors changes, the result of `getMainAuthor()` may h
...
+We can consider setting up a batch process that runs every few seconds
+to reindex all sensors, either through a <<mapper-orm-indexing-massindexer,mass indexer>>
+or <<mapper-orm-indexing-manual,other means>>.
+So we would really not mind if Hibernate Search just ignored changes to sensor values...
+
+That's what `@IndexingDependency(reindexOnUpdate = ReindexOnUpdate.NO)` is for:
+it tells Hibernate Search to ignore the impact of updates to the `rollingAverage` property.
+See the modified mapping below:
+
+.Disabling automatic reindexing for a particular property with `ReindexOnUpdate.NO`
```
Comment (fax4ever) [inline documentation/src/main/asciidoc/reference/mapper-orm-mapping-reindexing.asciidoc:245]: Same here. I think that it is not clear that this row is a title in the final rendering of the page.

**Review by fax4ever — APPROVED** [review]
Very nice job. Just few comments

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/automaticindexing/building/impl/PojoImplicitReindexingResolverBuildingHelper.java [hunk 2]
```diff
@@ -146,34 +146,32 @@ ContainerExtractorBinder getExtractorBinder() {
...
 
-	ReindexOnUpdate getReindexOnUpdate(ReindexOnUpdate parentReindexOnUpdate,
-			PojoTypeModel<?> typeModel, String propertyName, ContainerExtractorPath extractorPath) {
-		if ( ReindexOnUpdate.NO.equals( parentReindexOnUpdate ) ) {
-			return ReindexOnUpdate.NO;
+	ReindexOnUpdate getDefaultReindexOnUpdate() {
+		return defaultReindexOnUpdate;
+	}
+
+	ReindexOnUpdate getMetadataReindexOnUpdateOrNull(PojoTypeModel<?> typeModel,
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/automaticindexing/building/impl/PojoImplicitReindexingResolverBuildingHelper.java:153]: Yes, I couldn't find a better name. That's a map-like operation, so I guess `get` makes sense here.

File: documentation/src/main/asciidoc/reference/mapper-orm-mapping-reindexing.asciidoc [hunk 3]
```diff
@@ -116,37 +116,148 @@ that whenever the list of authors changes, the result of `getMainAuthor()` may h
...
 to a particular property (and, in the case of `@IndexedEmbedded`, anything beyond that property).
-The index will become slightly out-of-sync whenever the property is modified,
-but this can be solved by <<mapper-orm-indexing-massindexer,reindexing>>,
-for example every night.
 
-.Disabling automatic reindexing with `@IndexingDependency.reindexOnUpdate`
+Several options are available to control exactly how updates to a given property affect reindexing.
+See the sections below for an explanation of each option.
+
+=== `ReindexingOnUpdate.SHALLOW`: limiting automatic reindexing to same-entity updates only
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply documentation/src/main/asciidoc/reference/mapper-orm-mapping-reindexing.asciidoc:136]: Yes, you've said that before. In my opinion the issue is the CSS, not the backticks. I'll have to have a look someday.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**